File: prime_computing.py
import numpy as np


class PrimeComputing:
    def __innit__(self):
        pass


    def is_prime(self, number):
        # Check if a number is prime
        # number: integer
        # is_prime: boolean
        if number <= 1:
            return False
        for i in range(2, int(np.sqrt(number)) + 1):
            if number % i == 0:
                return False
        return True
    

    def get_primes(self, end, start=0):
        # Get all primes in a range
        # start: integer
        # end: integer
        # primes: list of integers
        primes = []
        for i in range(start + 1, end + 1):
            if self.is_prime(i):
                primes.append(i)
        return primes

File: cpp/src/prime_computing.cpp
#include <cmath>
#include "prime_computing.h"


bool PrimeComputing::isPrime(int number) {
    if (number <= 1) {
        return false;
    }
    for (int i = 2; i <= (int) sqrt(number); i++) {
        if (number % i == 0) {
            return false;
        }
    }
    return true;
}

std::vector<int> PrimeComputing::getPrimes(int end, int start) {
    std::vector<int> primes;
    for (int i = start; i <= end; i++) {
        if (this->isPrime(i)) {
            primes.push_back(i);
        }
    }
    return primes;
}